### DISCUSSION TITLE
installing language packages too slow

### DISCUSSION THREAD

**Discussion by tbagnoli** [explosion/spaCy#7374] [Help: Installation] [answered]
I'm aware that the issue has been [raised before](https://github.com/explosion/spaCy/issues/3998), but the proposed solution is just as slow.

## How to reproduce the problem
Trying to install a language package averages 10kb/s. The connection will time out and the install fail before that's finished.
```
pip install https://github.com/explosion/spacy-models/releases/download/de_core_news_sm-2.3.0/de_core_news_sm-2.3.0.tar.gz
```

In the issue above, it was suggested to download the package from a browser using the same link. The download speed from the browser is even slower, 2 kb/s! And no, this isn't down to my internet connection. A package that isn't quickly downloadable from pip is pretty much useless to anyone who really uses virtual environments and needs to create one often.

## Environment

* Operating System: MacOS 10.14.6 (18G6032)
* Python Version Used: 3.7.9
* spaCy Version Used: 2.2.4
* Environment Information: venv

**Comment by adrianeboyd** [comment, marked as answer]
The model downloads are github assets and github can be slow sometimes. All we can suggest is to wait a bit and try again, maybe checking https://www.githubstatus.com to see if there's any more info if it seems like it's not resolved relatively quickly.

If you are frequently downloading the same model packages, the solution is to download them once however you like (`wget`, browser, etc.) and then install them with `pip` from the local file in each new venv:

```python
pip install /path/to/de_core_news_sm-2.3.0.tar.gz
```
